.github/workflows/steelthread.yml

```diff
@@ -30,7 +30,12 @@ jobs:
       run: docker-compose -f "build/docker-compose.yml" up -d --build
 
     - name: End To End Test
-      run: mage integration
+      uses: nick-fields/retry@v2
+      with:
+        timeout_minutes: 15
+        max_attempts: 3
+        retry_on: error
+        command: mage integration
```

Comment:
I'm concerned that doing retrys would hide flakiness and real issues we should address. Thoughts?

Reply:
It does fail like 10% of the time, I think it has to do with the nondetermanistic threads test. I have retry logic on our client, but maybe there is some better handling we can do so the failure goes down to like 1% of the time

Reply:
removed this, yea lets keep it vanilla